tools/benchmark/src/main/java/com/datadog/benchmark/internal/reader/MemoryVitalReader.kt

```diff
@@ -7,16 +7,41 @@
 package com.datadog.benchmark.internal.reader
 
 import java.io.File
+import java.util.concurrent.Executors
+import java.util.concurrent.ScheduledExecutorService
+import java.util.concurrent.TimeUnit
 
 /**
  * Reads the device's `VmRSS` based on the `/proc/self/status` file.
  * cf. documentation https://man7.org/linux/man-pages/man5/procfs.5.html
  */
 internal class MemoryVitalReader(
-    internal val statusFile: File = STATUS_FILE
+    internal val statusFile: File = STATUS_FILE,
+    private val readerDelay: Long = STATUS_FILE_READ_INTERVAL_MS
 ) : VitalReader {
 
-    override fun readVitalData(): Double? {
+    private var maxMemory: Double = 0.0
+    private val executorService: ScheduledExecutorService = Executors.newSingleThreadScheduledExecutor()
+
+    private val readRunnable: Runnable = Runnable {
+        maxMemory = (readStatusFile() ?: 0.0).coerceAtLeast(maxMemory)
+    }
+
+    override fun readVitalData(): Double {
+        return maxMemory.apply {
+            maxMemory = 0.0
+        }
```

Comment:
this is kind of confusing syntax: `apply` has `this` as implicit receiver and the name means that we are doing something with receiver and returning it. But here nothing is done with receiver, instead modification is done to the property of `MemoryVitalReader`.

I think it will be better to write it without any sugar:

```
val current = maxMemory
maxMemory = 0.0
return current
```

Reply:
Agree, I will try to avoid confusing syntax sugar.